### PR TITLE
TemporaryFile: Implement move-assignment

### PR DESCRIPTION
**Pull request by dimitry-**
Test: Run OrbitBaseTests

### CODE REVIEW

File: src/OrbitBase/include/OrbitBase/TemporaryFile.h [hunk 1]
```diff
@@ -21,7 +21,16 @@ class TemporaryFile final {
     that.file_path_.clear();
   }
 
-  TemporaryFile& operator=(TemporaryFile&&) = delete;
+  TemporaryFile& operator=(TemporaryFile&& that) {
```
Comment (beckerhe) [inline src/OrbitBase/include/OrbitBase/TemporaryFile.h:24]: This can be made noexcept, can't it?

**Review by beckerhe — APPROVED** [review]
Otherwise LGTM

**Review by dpallotti — APPROVED** [review]
Thanks for addressing this.

File: src/OrbitBase/include/OrbitBase/TemporaryFile.h (hunk 1 above)
Comment (dimitry-) [reply src/OrbitBase/include/OrbitBase/TemporaryFile.h:24]: yes - done.

**Pull request merged**